### DISCUSSION TITLE
Multiple gauges from single expensive call

### DISCUSSION THREAD

**Discussion by ejsmith** [open-telemetry/opentelemetry-dotnet#3240] [Q&A]
I am attempting to populate queue stats like queue count, working item count and deadletter counts as 3 different observable gauges. Getting the counts requires a semi-expensive call and, when I make that, it returns all 3 counts at the same time. The problem is since gauges are required to be observable, I have to give different callbacks for each gauge. I am trying to figure out some sort of pattern where I can avoid having this semi-expensive call end up getting called 3 times in a row every time the values are observed. Does anyone know how to tackle this? Surely this shouldn't be so hard.

**Comment by alanwest** [comment]
There was a fairly recent addition to the metric API specification for addressing this use case: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#multiple-instrument-callbacks.

A few of the other language SDKs (Java and Go, I think) have already done some prototyping of this support. I'm pretty sure offering this support for .NET will require getting this on the .NET teams radar. @cijothomas do you think this issue is an appropriate place to raise this https://github.com/dotnet/runtime/issues/62027?

**Reply by ejsmith** [reply]
@alanwest thank you for the information. Yes, the multiple callbacks feature would solve the problem. @cijothomas, how do we attempt to raise the priority of this and try to get it included with .NET 7?
